compiler/src/dotty/tools/dotc/transform/init/Errors.scala

```diff
@@ -64,48 +64,51 @@ object Errors:
         else "^"
 
       s"$padding$carets\n"
-    }
 
     override def toString() = this.getClass.getName.nn
-  }
+  end Error
 
   /** Access non-initialized field */
-  case class AccessNonInit(field: Symbol, trace: Seq[Tree]) extends Error {
+  case class AccessNonInit(field: Symbol, trace: Seq[Tree]) extends Error:
     def source: Tree = trace.last
     def show(using Context): String =
       "Access non-initialized " + field.show + "."
 
     override def pos(using Context): SourcePosition = field.sourcePos
-  }
 
   /** Promote a value under initialization to fully-initialized */
-  case class PromoteError(msg: String, trace: Seq[Tree]) extends Error {
+  case class PromoteError(msg: String, trace: Seq[Tree]) extends Error:
     def show(using Context): String = msg
-  }
 
-  case class AccessCold(field: Symbol, trace: Seq[Tree]) extends Error {
+  case class AccessCold(field: Symbol, trace: Seq[Tree]) extends Error:
     def show(using Context): String =
       "Access field on a value with an unknown initialization status."
-  }
 
-  case class CallCold(meth: Symbol, trace: Seq[Tree]) extends Error {
+  case class CallCold(meth: Symbol, trace: Seq[Tree]) extends Error:
     def show(using Context): String =
       "Call method on a value with an unknown initialization" + "."
-  }
 
-  case class CallUnknown(meth: Symbol, trace: Seq[Tree]) extends Error {
+  case class CallUnknown(meth: Symbol, trace: Seq[Tree]) extends Error:
     def show(using Context): String =
       val prefix = if meth.is(Flags.Method) then "Calling the external method " else "Accessing the external field"
       prefix + meth.show + " may cause initialization errors" + "."
-  }
 
   /** Promote a value under initialization to fully-initialized */
-  case class UnsafePromotion(msg: String, trace: Seq[Tree], error: Error) extends Error {
+  case class UnsafePromotion(msg: String, trace: Seq[Tree], error: Error) extends Error:
     override def issue(using Context): Unit =
       report.warning(show, this.pos)
 
     def show(using Context): String =
-      msg + stacktrace + "\n" +
+      msg + stacktrace() + "\n" +
         "Promoting the value to fully initialized failed due to the following problem:\n" +
-        error.show + error.stacktrace
-  }
+        error.show + error.stacktrace()
+
+  /** Unsafe leaking a non-hot value as constructor arguments */
+  case class UnsafeLeaking(trace: Seq[Tree], error: Error) extends Error:
+    override def issue(using Context): Unit =
```

Comment:
The overrides of `issue` raise the question of what should be the difference between `show` and `issue`. The default `issue` combines the result of `show` with `stacktrace`, while these overrides remove the `stacktrace` and incorporate it into the result of `show`. Perhaps this should be made uniform, e.g. always incorporate the stack trace in `show` and never have to override `issue`.